lib/evmone/execution_state.hpp

```diff
@@ -83,6 +83,7 @@ class Memory
 
     uint8_t& operator[](size_t index) noexcept { return m_memory[index]; }
 
+    [[nodiscard]] const uint8_t* data() const noexcept { return m_memory.data(); }
```

Comment:
I wonder if in fact you don't actually want something like `bytes_view substr(size_t start, size_t size) const` here?

Reply:
There might be uses of such pattern, but tracing is not one of these.

Reply:
Well your test was doing this, what will be the use case in tracing?

Reply:
Dump all memory, e.g. `hex({memory.data(), memory.size()})`.